### PR TITLE
feat(cli): use `dialoguer` for interactive prompts (`login` & `delete`)

### PR DESCRIPTION
**Pull request by johnyob**
# Context

# Description

 **Dependencies**: #364 

This PR improves our current interactive prompts (for the account commands) by:
- Using a standardized library (dialoguer)
- Adds functionality to the prompts (account generation for `login`)

# Manually testing the PR

### CODE REVIEW

File: crates/jstz_cli/src/account.rs [hunk 1]
```diff
@@ -86,28 +86,15 @@ fn delete_account(alias: String) -> Result<()> {
...
-
-    info!("{}", confirmation_message);
-    io::stdout().flush()?;
+    if cfg.accounts.current_alias() == Some(&alias) {
+        warn!("You are currently logged into the account: {}.", alias);
+    }
 
-    let mut input = String::new();
-    io::stdin().read_line(&mut input)?;
+    let confirmation_alias: String = Input::new().with_prompt(format!("Are you sure you want to delete the account {0}? Please type '{0}' to confirm", alias)).interact()?;
```
Comment (sam-finch-tezos) [inline crates/jstz_cli/src/account.rs:93]: this is open to spelling errors, can it loop with either type the explicit confirmation or `n/no/abort` to abort? (unless of course the account is called `n` `no` or `abort`).

**Review by sam-finch-tezos — APPROVED** [review]
lgtm

File: crates/jstz_cli/src/account.rs (hunk 1 above)
Comment (johnyob) [reply crates/jstz_cli/src/account.rs:93]: True, but there is nothing stopping the user from pressing [Up Arrow] after the command exits to re-run the command

Comment (sam-finch-tezos) [reply crates/jstz_cli/src/account.rs:93]: Good enough for me 😃

**Review by alanmarkoTrilitech — APPROVED** [review]

**Pull request merged**